File: p118e/pascal_triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function (numRows) {
  let ans = [[1]];
  for (let r = 1; r < numRows; ++r) {
    let currRow = Array.from({ length: r + 1 }, (v) => 1);
    for (let i = 1; i < currRow.length - 1; ++i) {
      currRow[i] = ans[r - 1][i - 1] + ans[r - 1][i];
    }
    ans.push(currRow);
  }
  return ans;
};
// TEST
[1, 2, 3, 4, 5, 10].forEach((t) => {
  console.log("Pascal's triangle of order", t, '->');
  console.log(generate(t));
});

File: p118e/pascal_triangle.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        ans = [[1]]
        for n in range(1, numRows):
            ans.append(
                list(map(lambda a, b: a + b, [0] + ans[-1], ans[-1] + [0]))
            )
        return ans


for numRows, expected in [
    (1, [[1]]),
    (5, [[1], [1, 1], [1, 2, 1], [1, 3, 3, 1], [1, 4, 6, 4, 1]]),
]:
    sol = Solution()
    actual = sol.generate(numRows)
    print("Pascal's triangle of", numRows, "->", actual)
    assert actual == expected
